### PR TITLE
[webkitscmpy] Only respect the latest review

### PR DESCRIPTION
**Pull request by JonWBedard**
#### 3bced9d602295f75e2c637f4ccc6d1e733a6af0a
<pre>
[webkitscmpy] Only respect the latest review
<a href="https://bugs.webkit.org/show_bug.cgi?id=231987">https://bugs.webkit.org/show_bug.cgi?id=231987</a>
&lt;rdar://problem/84434991 &gt;

Reviewed by Dewei Zhu.

GitHub will report the entire history of a review, we should only
respect the latest review state.

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:
(GitHub.PRGenerator.reviewers):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/pull_request_unittest.py:
(TestNetworkPullRequestGitHub.test_approved_edits):
</pre>

### CODE REVIEW

**Comment by JonWBedard** [comment]
Test automatic comment

**Comment by JonWBedard** [comment]
Testing auto-closing

**Comment by JonWBedard** [comment]
Testing auto-closing

**Comment by JonWBedard** [comment]
Testing auto-reopening

**Comment by JonWBedard** [comment]
Sorry for the churn here, was using this bug to prototype `open`/`close` APIs

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py [hunk 1]
```diff
@@ -172,13 +172,17 @@ def reviewers(self, pull_request):
             pull_request._reviewers = [self._contributor(user['login']) for user in response.get('users', [])]
             pull_request._approvers = []
             pull_request._blockers = []
-            needs_status = Contributor.REVIEWER in self.repository.contributors.statuses
+
+            state_for = {}
             for review in self.repository.request('pulls/{}/reviews'.format(pull_request.number)):
-                contributor = self._contributor(review['user']['login'])
+                state_for[self._contributor(review['user']['login'])] = review.get('state')
```
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:212]: We assume reviews from `self.repository.request('pulls/{}/reviews'.format(pull_request.number))` are ordered by time from oldest to newest, right?

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py [hunk 2]
```diff
@@ -172,13 +172,17 @@ def reviewers(self, pull_request):
...
+
+            needs_status = Contributor.REVIEWER in self.repository.contributors.statuses
+            for contributor, status in state_for.items():
                 pull_request._reviewers.append(contributor)
-                if review.get('state') == 'APPROVED' and (not needs_status or contributor.status == Contributor.REVIEWER):
+                if status == 'APPROVED' and (not needs_status or contributor.status == Contributor.REVIEWER):
                     pull_request._approvers.append(contributor)
-                elif review.get('state') == 'CHANGES_REQUESTED':
+                elif status == 'CHANGES_REQUESTED':
```
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:219]: This whole change seems worth a unit test.

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py (hunk 1 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:212]: Yes!

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py (hunk 2 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:219]: We are already partially covered by unit tests, but I added a more explicit one.

Comment (dewei-zhu) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:219]: Great.

**Review by dewei-zhu — APPROVED** [review]

**Comment by JonWBedard** [comment]
Landed https://commits.webkit.org/243779@main (3332afeb2b4dba2cd80c812d73e0c81f63cbbf94)

**Pull request closed without merging**